stac_api/api/extensions/tiles_search.py

```diff
@@ -0,0 +1,231 @@
+        print(kwargs)
+        mosaic_assets = self.assets_for_tile(x, y, z, **kwargs)
+        if not mosaic_assets:
+            raise NoAssetFoundError(f"No assets found for tile {z}-{x}-{y}")
+
+        if reverse:
+            mosaic_assets = list(reversed(mosaic_assets))
+
+        # Sanitize kwargs to remove stac api kwargs
+        remove_kwargs = [
+            "collections",
+            "ids",
+            "datetime",
+            "limit",
+            "query",
+            "token",
+            "fields",
+            "sortby",
+        ]
+        kwargs = {k: v for k, v in kwargs.items() if k not in remove_kwargs}
+
+        def _reader(asset: str, x: int, y: int, z: int, **kwargs: Any) -> ImageData:
+            # Take out reader_options for now because that's how I'm passing the
+            # postgres client to self.get_assets
+            # with self.reader(asset, **self.reader_options) as src_dst:
+            with self.reader(asset) as src_dst:
+                return src_dst.tile(x, y, z, **kwargs)
+
+        return mosaic_reader(mosaic_assets, _reader, x, y, z, **kwargs)
+
+
+@dataclass
+class AssetParams(DefaultDependency):
+    """Band names and Expression parameters."""
+
+    assets: Optional[str] = Query(
+        None,
+        title="Asset names",
+        description="comma-delimited asset names.",
+    )
+
+    def __post_init__(self):
+        """Post Init."""
+        if self.assets is not None:
+            self.kwargs["assets"] = self.assets.split(",")
+
+
+@dataclass
+class DatasetDependency(DefaultDependency):
+    """Add STAC search params to query."""
+
+    collections: Optional[str] = Query(
+        None, title="Collections", description="Collections to search within"
+    )
+
+    ids: Optional[str] = Query(None)
+    datetime: Optional[str] = Query(None)
+    limit: Optional[int] = Query(10)
+    query: Optional[str] = Query(None)
+    token: Optional[str] = Query(None)
+    fields: Optional[str] = Query(None)
+    sortby: Optional[str] = Query(None)
+
+    def __post_init__(self):
+        """Post Init."""
+        if self.collections:
+            self.kwargs["collections"] = self.collections.split(",")
+        if self.ids:
+            self.kwargs["ids"] = self.ids.split(",")
+        if self.datetime:
+            self.kwargs["datetime"] = self.datetime
+        if self.limit:
+            self.kwargs["limit"] = self.limit
+        if self.query:
+            self.kwargs["query"] = self.query
+        if self.token:
+            self.kwargs["token"] = self.token
+        if self.fields:
+            self.kwargs["fields"] = self.fields.split(",")
+        if self.sortby:
+            self.kwargs["sortby"] = self.sortby.split(",")
+
+
+@attr.s
+class TilesSearchExtension(ApiExtension):
+    """Custom Titiler Extension.
+
+    Custom extension for sending query results to a titiler DynamicSTAC endpoint
+    """
+
+    client: CoreCrudClient = attr.ib(default=attr.Factory(CoreCrudClient))
```

Comment:
This uses the base `CoreCrudClient` as all it needs is the `client.get_search` method. I figured it wasn't necessary to subclass `CoreCrudClient` like the `TilesClient` does

https://github.com/arturo-ai/arturo-stac-api/blob/7ca0c8790ac64bbd359e88933ef94f25ece5b383/stac_api/clients/tiles/ogc.py#L17-L41

Reply:
makes sense.  `TilesClient` arguably shouldn't subclass `CoreCrudClient` either.  I think composition is a better pattern here.